### PR TITLE
Config filters

### PR DESCRIPTION
**Pull request by bolzo**
The predefined filter are case insensitive.
Filtered data is not purged but Masked with `** HIDDEN **`
Configs can be logged with `--log-gathered-config`

```
ubuntu@mon1:~/ceph-collect$ ./ceph-collect --log-gathered-config --config-filter rgw
INFO:root:Gathering overall system information
No LSB modules are available.
INFO:root:Gathering overall Ceph information
INFO:root:==== ceph.conf ======
INFO:root: - # Please do not change this file directly since it is managed by Ansible and will be overwritten
INFO:root: - [global]
INFO:root: - cluster network = 172.16.4.0/24
INFO:root: - fsid = 35927808-bc77-444c-8f40-489e78c474fa
INFO:root: - mon host = [v2:172.16.4.45:3300,v1:172.16.4.45:6789],[v2:172.16.4.113:3300,v1:172.16.4.113:6789],[v2:172.16.4.130:3300,v1:172.16.4.130:6789]
INFO:root: - mon initial members = mon1,mon2,mon3
INFO:root: - osd pool default crush rule = -1
INFO:root: - public network = 172.16.4.0/24
INFO:root: -
INFO:root: - [osd]
INFO:root: - bluestore_min_alloc_size_hdd = 4096
INFO:root: -
INFO:root: - [mgr]
INFO:root: - mgr/alerts/smtp_password  = ** HIDDEN **
INFO:root: - mgr/alerts/smtp_PaSSwOrd  = ** HIDDEN **
INFO:root:====== config ======
INFO:root: - WHO             MASK  LEVEL     OPTION                                          VALUE       RO
INFO:root: - global                advanced  mon_allow_pool_delete                           true
INFO:root: - global                advanced  mon_max_pg_per_osd                              2000
INFO:root: - global                advanced  osd_pool_default_pg_autoscale_mode              warn
INFO:root: -   mon                 advanced  mon_warn_on_insecure_global_id_reclaim          false
INFO:root: -   mon                 advanced  mon_warn_on_insecure_global_id_reclaim_allowed  false
INFO:root: -   mgr                 advanced  mgr/alerts/smtp_password                        ** HIDDEN **
INFO:root: -   mgr                 advanced  mgr/alerts/smtp_user                            test
INFO:root: -   mgr                 advanced  mgr/balancer/active                             true
INFO:root: -   mgr                 advanced  mgr/balancer/mode                               upmap
INFO:root: -   mgr                 advanced  mgr/influx/password                             ** HIDDEN **    *
INFO:root: -   mgr                 advanced  mgr/influx/username                             admin123    *
INFO:root: -   mgr                 basic     target_max_misplaced_ratio                      0.200000
INFO:root: -   osd                 advanced  osd_max_backfills                               12
INFO:root: -   osd                 basic     osd_memory_target                               1073741824
INFO:root: -   osd                 advanced  osd_recovery_max_active                         16
INFO:root: -   osd                 advanced  osd_recovery_sleep_hdd                          0.000000
INFO:root: -   osd                 advanced  osd_scrub_during_recovery                       true
INFO:root: -     client.rgw        basic     rgw_dynamic_resharding                          ** HIDDEN **
INFO:root: -     client.rgw        advanced  rgw_max_dynamic_shards                          ** HIDDEN **
INFO:root: -     client.rgw        basic     rgw_max_objs_per_shard                          ** HIDDEN **
INFO:root: -     client.rgw        advanced  rgw_reshard_thread_interval                     ** HIDDEN **
INFO:root: -     client.rgw        advanced  rgw_safe_max_objects_per_shard                  ** HIDDEN **
INFO:root: -     client.rgw        advanced  rgw_shard_warning_threshold                     ** HIDDEN **
INFO:root:Gathering Health information
INFO:root:Gathering MON information
INFO:root:Gathering OSD information
INFO:root:Gathering PG information
INFO:root:Gathering MDS information
INFO:root:Outputted Ceph information to /tmp/ceph-collect_20220218_130143.tar.gz
```

### CODE REVIEW

File: ceph-collect [hunk 1]
```diff
@@ -279,20 +290,32 @@ def collect_ceph_information(r, ceph_config, output_directory, timeout,
...
+                True
             )
         )
     files['config'] = filter_config(
         ceph_mon_command(r, 'config dump', timeout, output_format),
-        output_format
+        output_format,
+        False
     )
+    if log_config:
```
Comment (rubenk) [inline ceph-collect:310]: While this is nice, it's not strictly needed to fix the issue. Let's keep ceph-collect simple and not extend it with command line options. If you want to know what's in ceph.conf, you can just extract it from the tarball.

File: ceph-collect [hunk 2]
```diff
@@ -223,30 +229,35 @@ def collect_ceph_information(r, ceph_config, output_directory, timeout,
                     * 'plain' 
                     * 'json'   
             :type mode: ``str``
+            :param is_conffile:
+                True is data is from "ceph.conf" (INI file)
```
Comment (rubenk) [inline ceph-collect:233]: Pleas fix the typo and remove (INI file), that doesn't add any information.

File: ceph-collect [hunk 3]
```diff
@@ -223,30 +229,35 @@ def collect_ceph_information(r, ceph_config, output_directory, timeout,
...
 
             lines=data.splitlines()
             for patter in config_filters:
-                for index in range(len(lines)-1,-1,-1):
-                    if bool(re.search(patter, lines[index])):
-                        del lines[index]
+                for index in range(len(lines)-1, -1, -1):
+                    line=lines[index]
+                    if bool(re.search(patter, line)):
+                        if is_conffile:
```
Comment (rubenk) [inline ceph-collect:248]: can you explain the logic here, in a comment? Why are config files different?

File: ceph-collect [hunk 4]
```diff
@@ -279,20 +290,32 @@ def collect_ceph_information(r, ceph_config, output_directory, timeout,
         files['fsid'] = bytes(r.get_fsid() + '\n', 'utf-8')
         files['ceph.conf'] = filter_config(
                 get_ceph_config(ceph_config), 
-                'plain'
+                'plain',
+                True
```
Comment (rubenk) [inline ceph-collect:294]: Please use a keyword arg here (so is_conffile=True)

File: ceph-collect (hunk 1 above)
Comment (bolzo) [reply ceph-collect:319]: I think you need it if you want to filter your config: save it and extract the data has two problems, 1 is boring and 2 you need to know which file is. 
We implement this to allow people to filter the data. If they need to extract the data, they can filter it by themself.

**Pull request merged**